packages/plugin-ext/src/main/browser/scm-main.ts

```diff
@@ -37,12 +37,30 @@ export class ScmMainImpl implements ScmMain {
     private readonly scmService: ScmService;
     private readonly scmRepositoryMap: Map<number, ScmRepository>;
     private readonly labelProvider: LabelProvider;
+    private lastSelectedSourceControlHandle: number | undefined;
 
     constructor(rpc: RPCProtocol, container: interfaces.Container) {
         this.proxy = rpc.getProxy(MAIN_RPC_CONTEXT.SCM_EXT);
         this.scmService = container.get(ScmService);
         this.scmRepositoryMap = new Map();
         this.labelProvider = container.get(LabelProvider);
+        this.scmService.onDidChangeSelectedRepository(repository => {
+            const sourceControlHandle = repository ? this.getSourceControlHandle(repository) : undefined;
```

Comment:
maybe we are better to extract a method `updateSelectedRepository` which calls `this.scmService.selectedRepository`

When we call it initially and from event handler.

Reply:
done